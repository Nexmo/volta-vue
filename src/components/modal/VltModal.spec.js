import { shallowMount } from '@vue/test-utils';
import { VltModal } from '..';

describe('vlt-modal', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(VltModal, {
      propsData: {
        title: 'Title',
      },
    });
  });

  test('can be opened', () => {
    wrapper.setProps({ visible: true });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('renders correctly with title "Title"', () => {
    wrapper.setProps({ visible: true });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is a message modal', () => {
    wrapper.setProps({
      message: true,
      visible: true,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('has customer Ok button text "Delete"', () => {
    wrapper.setProps({
      okText: 'Delete',
      visible: true,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('can be cancelled', () => {
    wrapper.setProps({ visible: true });
    wrapper.findAll('button').at(0).trigger('click');
    expect(wrapper.emitted('close')).toBeTruthy();
  });

  test('can be confirmed', () => {
    wrapper.setProps({ visible: true });
    wrapper.findAll('button').at(1).trigger('click');
    expect(wrapper.emitted('confirm')).toBeTruthy();
  });

  test('modal without footer', () => {
    wrapper.setProps({
      noFooter: true,
      visible: true,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('can be closed', () => {
    wrapper.setProps({ visible: true });
    wrapper.find('.Vlt-modal__dismiss').trigger('click');
    expect(wrapper.emitted('close')).toBeTruthy();
  });
});
