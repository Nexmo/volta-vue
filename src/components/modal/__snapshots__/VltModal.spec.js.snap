// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-modal can be opened 1`] = `
<div
  class="Vlt-modal Vlt-modal_visible"
>
  <div
    class="Vlt-modal__panel"
  >
    <header
      class="Vlt-modal__header"
    >
      <h4>
        Title
      </h4>
       
      <a
        aria-label="Close"
        class="Vlt-modal__dismiss"
      />
    </header>
     
    <section
      class="Vlt-modal__content"
    >
      <p>
        
      </p>
    </section>
     
    <footer
      class="Vlt-modal__footer"
    >
      <!---->
       
      <button
        class="Vlt-btn Vlt-btn--app Vlt-btn--tertiary"
      >
        
          Cancel
        
      </button>
       
      <button
        class="Vlt-btn Vlt-btn--app Vlt-btn--secondary"
      >
        <!---->
        
          Ok
        
      </button>
    </footer>
  </div>
</div>
`;

exports[`vlt-modal can footer be hidden 1`] = `<!---->`;

exports[`vlt-modal has customer Ok button text "Delete" 1`] = `
<div
  class="Vlt-modal Vlt-modal_visible"
>
  <div
    class="Vlt-modal__panel"
  >
    <header
      class="Vlt-modal__header"
    >
      <h4>
        Title
      </h4>
       
      <a
        aria-label="Close"
        class="Vlt-modal__dismiss"
      />
    </header>
     
    <section
      class="Vlt-modal__content"
    >
      <p>
        
      </p>
    </section>
     
    <footer
      class="Vlt-modal__footer"
    >
      <!---->
       
      <button
        class="Vlt-btn Vlt-btn--app Vlt-btn--tertiary"
      >
        
          Cancel
        
      </button>
       
      <button
        class="Vlt-btn Vlt-btn--app Vlt-btn--secondary"
      >
        <!---->
        
          Delete
        
      </button>
    </footer>
  </div>
</div>
`;

exports[`vlt-modal is a message modal 1`] = `
<div
  class="Vlt-modal Vlt-modal_visible Vlt-modal--message"
>
  <div
    class="Vlt-modal__panel"
  >
    <header
      class="Vlt-modal__header"
    >
      <h4>
        Title
      </h4>
       
      <a
        aria-label="Close"
        class="Vlt-modal__dismiss"
      />
    </header>
     
    <section
      class="Vlt-modal__content"
    >
      <p>
        true
      </p>
    </section>
     
    <footer
      class="Vlt-modal__footer"
    >
      <!---->
       
      <button
        class="Vlt-btn Vlt-btn--app Vlt-btn--tertiary"
      >
        
          Cancel
        
      </button>
       
      <button
        class="Vlt-btn Vlt-btn--app Vlt-btn--secondary"
      >
        <!---->
        
          Ok
        
      </button>
    </footer>
  </div>
</div>
`;

exports[`vlt-modal renders correctly with title "Title" 1`] = `
<div
  class="Vlt-modal Vlt-modal_visible"
>
  <div
    class="Vlt-modal__panel"
  >
    <header
      class="Vlt-modal__header"
    >
      <h4>
        Title
      </h4>
       
      <a
        aria-label="Close"
        class="Vlt-modal__dismiss"
      />
    </header>
     
    <section
      class="Vlt-modal__content"
    >
      <p>
        
      </p>
    </section>
     
    <footer
      class="Vlt-modal__footer"
    >
      <!---->
       
      <button
        class="Vlt-btn Vlt-btn--app Vlt-btn--tertiary"
      >
        
          Cancel
        
      </button>
       
      <button
        class="Vlt-btn Vlt-btn--app Vlt-btn--secondary"
      >
        <!---->
        
          Ok
        
      </button>
    </footer>
  </div>
</div>
`;
